src/shogun/kernel/PeriodicKernel.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2015 Esben Soerig
+ */
+
+#ifndef PERIODICKERNEL_H
+#define PERIODICKERNEL_H
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/kernel/Kernel.h>
+#include <shogun/kernel/DotKernel.h>
+#include <shogun/features/DotFeatures.h>
+
+namespace shogun
+{
+	class CDotFeatures;
+
+/** @brief The periodic kernel as described in <i>The Kernel Cookbook</i>
+ * by David Duvenaud
+ *
+ * It is computed as
+ *
+ * \f[
+ *   k({\bf x},{\bf x'})= exp(-\frac{2sin^2(\pi|{\bf x}-{\bf x'}|/p)}{l^2})
+ * \f]
+ *
+ * where \f$p\f$ is the period of the kernel and \f$l\f$ is the length scale
+ * (width) of the kernel.
+ */
+
+class CPeriodicKernel: public CDotKernel
+{
+	public:
+		/** default constructor */
+		CPeriodicKernel();
+
+		/** constructor
+		 *
+		 * @param size cache size
+		 * @param width width
+		 * @param period period
+		 */
+		CPeriodicKernel(int32_t size, float64_t width, float64_t period);
```

Comment:
could you offer a constructor that hides the cache size and sets it to a default value. I always thought it was annoying to choose it ;)